modules/nw-http-header-configuration.adoc

```diff
@@ -0,0 +1,135 @@
+====
+
+== Order of precedence
+
+When the same HTTP header is modified both in the Ingress Controller and route object, HAProxy prioritizes the actions in certains ways depending on if it is a request or response header.
+
+* In the case of HTTP response headers, actions specified in the Ingress Controller are executed after the actions specified in a route. This means that the actions specified in the Ingress Controller take precedence.
+
+* In the case of HTTP request headers, actions specified in a route object are executed after the actions specified in the Ingress Controller. This means that the actions specified in the route object take precedence.
+
+For example, if a cluster admin sets the X-Frame-Options response header with the value `DENY` in the Ingress Controller:
+
+.Example Ingress Controller spec
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: DENY
+----
+
+And if a route owner sets the same X-Frame-Options response header with the value `SAMEORIGIN`:
+
+.Example route spec
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: SAMEORIGIN
+----
+
+Then the value set for the X-Frame-Options header at the global level in the Ingress Controller will take precedence, even if a specific route allows frames.
+
+This is because `haproxy.config` uses the following logic, where the Ingress Controller is considered the frontend and individual routes are considered the backend. The header value `DENY` applied to the frontend configurations overrides the same header with the value `SAMEORIGIN` that is set in the backend section:
+
+[source,text]
+----
+frontend public
+  http-response set-header X-Frame-Options 'DENY'
+
+frontend fe_sni
+  http-response set-header X-Frame-Options 'DENY'
+
+frontend fe_no_sni
+  http-response set-header X-Frame-Options 'DENY'
+
+backend be_secure:openshift-monitoring:alertmanager-main
+  http-response set-header X-Frame-Options 'SAMEORIGIN'
+----
+
+Additionally, any actions defined in either the Ingress Controller or a route object override values set using route annotations.
+
+== Special case headers
+
+The following headers are either prevented entirely from being set or deleted, or allowed under specific circumstances:
+
+.Header configuration options
+[cols="1,1,1,80,15a",options="header"]
```

Comment:
It might be unavoidable, but this table isn't presented very well. With having to scroll, and some longer items wrapping. Columns 2 and 3 are taking up way too much space just because of the header.

My recommendations would be to try to be more concise with the headers in columns 2 and 3. Probably removing "spec.httpHeaders.actions" would help, since it's the same for both. If it's important, could update the title for the table to include it

And then consider combining columns 4 and 5. I think 5 columns is too many to expect them to render well. I'm sure you could find a title that encompasses both of these

Reply:
yeah i've tried every which way with the table, it seems to be an asciibinder processing issue 🙁 i filed an issue: https://github.com/redhataccess/ascii_binder/issues/184

i will at least clean up the headers since they're not respected anyway